addons/mail/static/src/component_hooks/use_ref_to_model/use_ref_to_model.js

```diff
@@ -0,0 +1,44 @@
+/** @odoo-module **/
+
+import { clear } from '@mail/model/model_field_command';
+
+const { onWillUpdateProps, useComponent, useRef } = owl.hooks;
+
+/**
+ * This hook provides support for saving the result of useRef directly into the
+ * field of a record, and appropriately updates it when necessary (props change
+ * or destroy).
+ *
+ * @param {Object} param0
+ * @param {string} param0.fieldName Name of the field on the target record.
+ * @param {string} param0.modelName Name of the model of the target record.
+ * @param {string} param0.localIdPropName Name of the prop of this component
+ *  containing the localId of the target record.
+ * @param {string} param0.refName Name of the t-ref on this component.
+ */
+export function useRefToModel({ fieldName, modelName, localIdPropName, refName }) {
```

Comment:
I suggest `propNameAsRecordLocalId`. It's more natural and less obscure than current name.

Reply:
Ok to rename